--- src/Terraria/Terraria/GameContent/ShopHelper.cs
+++ src/Tea/Terraria/GameContent/ShopHelper.cs
@@ -4,6 +_,7 @@
 using Terraria.GameContent.Personalities;
 using Terraria.ID;
 using Terraria.Localization;
+using Terraria.Tea;
 
 namespace Terraria.GameContent
 {
@@ -106,7 +_,7 @@
 					_currentPriceAdjustment *= 0.9f;
 				}
 
-				bool[] array = new bool[663];
+				bool[] array = new bool[NPCLoader.Count];
 				foreach (NPC item in nearbyResidentNPCs) {
 					array[item.type] = true;
 				}
