--- src/Terraria/Terraria/MessageBuffer.cs
+++ src/Tea/Terraria/MessageBuffer.cs
@@ -17,6 +_,7 @@
 using Terraria.ID;
 using Terraria.Localization;
 using Terraria.Net;
+using Terraria.Tea;
 using Terraria.Testing;
 using Terraria.UI;
 
@@ -1141,7 +_,7 @@
 							Item item2 = Main.item[num72];
 							ItemSyncPersistentStats itemSyncPersistentStats = default(ItemSyncPersistentStats);
 							itemSyncPersistentStats.CopyFrom(item2);
-							bool newAndShiny = (item2.newAndShiny || item2.netID != num71) && ItemSlot.Options.HighlightNewItems && (num71 < 0 || num71 >= 5045 || !ItemID.Sets.NeverAppearsAsNewInInventory[num71]);
+							bool newAndShiny = (item2.newAndShiny || item2.netID != num71) && ItemSlot.Options.HighlightNewItems && (num71 < 0 || num71 >= ItemLoader.Count || !ItemID.Sets.NeverAppearsAsNewInInventory[num71]);
 							item2.netDefaults(num71);
 							item2.newAndShiny = newAndShiny;
 							item2.Prefix(pre2);
@@ -1319,7 +_,7 @@
 						if (num83 == 245)
 							NPC.golemBoss = num80;
 
-						if (nPC3.type >= 0 && nPC3.type < 663 && Main.npcCatchable[nPC3.type])
+						if (nPC3.type >= 0 && nPC3.type < NPCLoader.Count && Main.npcCatchable[nPC3.type])
 							nPC3.releaseOwner = reader.ReadByte();
 
 						break;
@@ -2163,7 +_,7 @@
 						if (Main.netMode != 2)
 							break;
 
-						if (num179 >= 0 && num179 < 663 && NPCID.Sets.MPAllowedEnemies[num179]) {
+						if (num179 >= 0 && num179 < NPCLoader.Count && NPCID.Sets.MPAllowedEnemies[num179]) {
 							if (!NPC.AnyNPCs(num179))
 								NPC.SpawnOnPlayer(plr, num179);
 						}
