--- src/Terraria/Terraria/Terraria.csproj
+++ src/Tea/Terraria/Terraria.csproj
@@ -18,7 +_,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <PdbFile>bin\$(OutputName)</PdbFile>
-    <ApplicationIcon>Icon.ico</ApplicationIcon>
+    <ApplicationIcon>Tea.ico</ApplicationIcon>
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="FNA" Condition="$(DefineConstants.Contains('FNA'))" />
@@ -62,8 +_,8 @@
     <EmbeddedResource Remove="Libraries/Mac/**" Condition="!$(DefineConstants.Contains('MAC'))" />
     <EmbeddedResource Remove="Libraries/Linux/**" Condition="!$(DefineConstants.Contains('LINUX'))" />
     <EmbeddedResource Remove="Libraries/Mono/**" Condition="!$(DefineConstants.Contains('MONO'))" />
-    <EmbeddedResource Remove="Libraries/XNA/**"/>
+    <EmbeddedResource Remove="Libraries/XNA/**" />
-    <EmbeddedResource Remove="Libraries/FNA/**"/>
+    <EmbeddedResource Remove="Libraries/FNA/**" />
     <EmbeddedResource Include="Localization/Content/**" />
     <EmbeddedResource Include="Microsoft/**" />
   </ItemGroup>
